go/chat/sender.go

```diff
@@ -86,28 +87,39 @@ func (s *BlockingSender) addPrevPointersToMessage(ctx context.Context, msg chat1
 
 	var prevs []chat1.MessagePreviousPointer
 
-	res, err := s.G().ConvSource.PullLocalOnly(ctx, convID, msg.ClientHeader.Sender,
+	res, _, err := s.G().ConvSource.Pull(ctx, convID, msg.ClientHeader.Sender,
 		&chat1.GetThreadQuery{
 			DisableResolveSupersedes: true,
 		},
 		&chat1.Pagination{
-			Num: -1,
+			Num: 50,
 		})
-	switch err.(type) {
-	case storage.MissError:
-		s.Debug(ctx, "No local messages; skipping prev pointers")
-	case nil:
-		if len(res.Messages) == 0 {
-			s.Debug(ctx, "no local messages found for prev pointers")
-		}
-		prevs, err = CheckPrevPointersAndGetUnpreved(&res)
-		if err != nil {
-			return chat1.MessagePlaintext{}, err
-		}
-	default:
+	if err != nil {
 		return chat1.MessagePlaintext{}, err
 	}
 
+	if len(res.Messages) == 0 {
+		s.Debug(ctx, "no local messages found for prev pointers")
+	}
+	prevs, err = CheckPrevPointersAndGetUnpreved(&res)
+	if err != nil {
+		return chat1.MessagePlaintext{}, err
+	}
+
+	if len(prevs) == 0 {
+		return chat1.MessagePlaintext{}, fmt.Errorf("Could not find previous messsages for prev pointers (of %v)", len(res.Messages))
+	}
+
+	for _, msg2 := range res.Messages {
+		if msg2.IsValid() {
+			err = s.checkConvID(ctx, convID, msg, msg2)
+			if err != nil {
+				return chat1.MessagePlaintext{}, err
+			}
+			break
```

Comment:
Added this break, so that it only checks against one message instead of up to 50.